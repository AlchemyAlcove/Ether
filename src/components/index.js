import Button from "./button";
import Ellipsis from "./Ellipsis";

export{
  Button,
  Ellipsis
};
