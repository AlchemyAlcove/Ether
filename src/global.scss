$containerWidth: 1200px;
$desktopBreakpoint: 992px;
$headerHeight: 50px;
$tabletBreakpoint: 768px;

$background: #ECF0F1;
$headerBackground: #363D44;
$headerBackgroundHighlight: #24292D;
$headerDropdownBackground: #FFFFFF;
$headerLogoHighlight: #1EB6F2;
$headerText: #FFFFFF;
$headerTextOpaque: #BFC4CA;
$text: #333333;

$font: 'Open Sans', sans-serif;
$weightThin: 300;
$weightRegular: 400;
$weightThick: 700;

body, html {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
  background: $background;
  font-size: 14px;
  font-family: $font;
  font-weight: $weightRegular;
  color: $text;

  * {
    box-sizing: border-box;
  }
}

*:focus {
  outline: 0;
}

a {
  color: $text;
  text-decoration: none;
  cursor: pointer;
}

@import "./components/ellipsis/ellipsis.scss";