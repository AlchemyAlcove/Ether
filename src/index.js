import Ellipsis from "./components/ellipsis";
import Inject from "./utils/inject";

export{
  Ellipsis,
  Inject
};
