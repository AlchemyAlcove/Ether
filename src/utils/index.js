import setVariables from "./setVariables";

export{
  setVariables
};
