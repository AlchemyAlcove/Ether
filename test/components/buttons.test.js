import { Button } from "../../src";
import { fireEvent, render } from "react-testing-library";
import React from "react"; // eslint-disable-line no-unused-vars
import sinon from "sinon";

test("disabled generic button", () => {
  <Button disabled={true}/>
  const{container} = render(<Button disabled={true}/>);

  expect(container.firstChild).toMatchSnapshot();
});

test("secondary large button", () => {
  <Button disabled={true}/>
  const{container} = render(<Button size="lg" color="secondary"/>);

  expect(container.firstChild).toMatchSnapshot();
});

test("success small outline button", () => {
  <Button disabled={true}/>
  const{container} = render(<Button size="sm" color="success" type="outline"/>);

  expect(container.firstChild).toMatchSnapshot();
});

test("danger small link button", () => {
  <Button disabled={true}/>
  const{container} = render(<Button size="sm" color="danger" type="link"/>);

  expect(container.firstChild).toMatchSnapshot();
});

test("warning small button", () => {
  <Button disabled={true}/>
  const{container} = render(<Button size="sm" color="warning"/>);

  expect(container.firstChild).toMatchSnapshot();
});

test("click", () => {
  const onClickSpy = sinon.spy();
  const{container} = render(<Button onClick={onClickSpy}/>);

  fireEvent.click(container.firstChild);
  sinon.assert.calledOnce(onClickSpy);
});

test("disabled click", () => {
  const onClickSpy = sinon.spy();
  const{container} = render(<Button disabled={true} onClick={onClickSpy}/>);

  fireEvent.click(container.firstChild);
  sinon.assert.notCalled(onClickSpy);
});