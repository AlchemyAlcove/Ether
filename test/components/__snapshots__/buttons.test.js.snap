// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`danger small link button 1`] = `
<button
  class="color-danger size-sm type-link "
  type="button"
/>
`;

exports[`disabled generic button 1`] = `
<button
  class="color-primary size-md type-solid "
  disabled=""
  type="button"
/>
`;

exports[`secondary large button 1`] = `
<button
  class="color-secondary size-lg type-solid "
  type="button"
/>
`;

exports[`success small outline button 1`] = `
<button
  class="color-success size-sm type-outline "
  type="button"
/>
`;

exports[`warning small button 1`] = `
<button
  class="color-warning size-sm type-solid "
  type="button"
/>
`;
